extensions/ext/GLSL_EXT_spirv_intrinsics.txt

```diff
@@ -0,0 +1,1020 @@
+        *OpDecorateString* is.
+
+        Any extensions or capabilities needed by the decoration must be
+        declared using SPIR-V Requirements Declarations (see section 3.9).
+
+        Multiple `spirv_decoration` qualifiers can be added to a single
+        declaration or expression.
+
+    Add a new section 4.14. SPIR-V Types
+
+        SPIR-V types not known to or exposed by the GLSL compiler can
+        be added by use of the `spirv_type` specifier.
+
+        This functionality is only available when generating SPIR-V.
+
+        The `spirv_type()` specifier can be used in place of a type name
+        for variable declarations.
+
+        spirv_type-specifier:
+
+            `spirv_type` ( spirv_instruction-qualifier-list )
+            `spirv_type` ( spirv_instruction-qualifier-list, spirv_type-parameter-list )
+            `spirv_type` ( spirv-requirements-list, spirv_instruction-qualifier-list )
+            `spirv_type` ( spirv-requirements-list, spirv_instruction-qualifier-list, spirv_type-parameter-list )
+
+        spirv_type-parameter-list:
+
+            spirv_type-parameter
+            spirv_type-parameter , spirv_type-parameter-list
+
+        spirv_type-parameter:
+
+            constant-expression
+            literal
+
+        spirv_instruction-qualifier-list:
+
+            spirv_instruction-qualifier
+            spirv_instruction-qualifier , spirv_instruction-qualifier-list
+
+        spirv_instruction-qualifier:
+
+            set = literal-string
+            id = integer-constant-expression
+
+        The `set` qualifier indicates the name of an extended
+        instruction set that the type's instruction belongs to.
+        If the `set` qualifier is omitted, or the string is empty,
+        no extended instruction set will be imported, and the instruction will
+        be selected from the core instruction set.
+
+        The `id` is the id of the type instruction in the selected
+        instruction set.
+
+        Parameters to this qualifier must be either literal values, or constant
+        expressions as defined in section 4.3.3, Constant Expressions.
+
+        When generating SPIR-V, the compiler will generate type
+        instructions for these types in the following manner:
+
+          * If any variable is declared with a SPIR-V type that requires an
+            extended instruction set (declared with the `set` qualifier),
+            it will add an *OpExtInstImport* instruction with the same
+            literal string.
+          * For variables declared with a SPIR-V type that requires an
+            extended instruction set, an *OpExtInst* instruction will be
+            inserted into the resulting SPIR-V, declaring the type.
+          * For variables declared with a SPIR-V type that is defined by
+            instructions in the core instruction set, an instruction with
+            the declared `id` will be inserted into the resulting SPIR-V.
```

Comment:
"an instruction with the declared id": Which type of instruction? The other 2 bullet points say this explicitly so this sounds a bit odd.

Reply:
I wasn't really looking for constraints as much as explanation. I think my confusion comes from the fact that the ID here is actually the opcode of the instructions that declares the type? I had thought that it was talking about the result-id.

Reply:
For now I'll just put in "...with the declared `id` as its opcode will be inserted.." but happy to iterate further.